lib/codecept.js

```diff
@@ -186,14 +185,31 @@ class Codecept {
     }
 
     if (this.opts.shuffle) {
-      this.testFiles = shuffle(this.testFiles)
+      this.testFiles = this.shuffle(this.testFiles)
     }
 
     if (this.opts.shard) {
       this.testFiles = this._applySharding(this.testFiles, this.opts.shard)
     }
   }
 
+  /**
+   * Fisher-Yates algorithm for shuffle
+   */
+  shuffle(arrayToShuffle) {
+    var i = arrayToShuffle.length,
+      j,
+      temp
+    while (--i > 0) {
+      j = Math.floor(Math.random() * (i + 1))
+      temp = arrayToShuffle[j]
+      arrayToShuffle[j] = arrayToShuffle[i]
+      arrayToShuffle[i] = temp
+    }
+
+    return arrayToShuffle
+  }
+
```

Comment:
@thomashohn what do you think about this? Not from mine but from you knew where already.

```
/**
 * Fisher–Yates shuffle (non-mutating)
 */
function shuffle(array) {
  const arr = [...array]; // clone to avoid mutating input
  for (let i = arr.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [arr[i], arr[j]] = [arr[j], arr[i]]; // swap
  }
  return arr;
}

```

- Uses modern let, const, and destructuring.
- Doesn’t modify the original array (important for functional/pure logic).
- Clearer loop (for instead of while (--i > 0)).